packages/react-native/UPGRADING.md

```diff
@@ -0,0 +1,279 @@
+
+To facilitate this, and to get Bugsnag started as early as possible, most configuration can only be done in the native layer. Some JS-specific options are allowed in the JS layer.
+
+##### iOS
+
+Update any of the following imports in. These will likely be in `AppDelegate.m` where it's recommended to configure Bugsnag, but may be in other locations too:
+
+```diff
+- #import <BugsnagReactNative/BugsnagReactNative.h>
+- #import "BugsnagConfiguration.h"
++ import <Bugsnag/Bugsnag.h>
+```
+
+Inside your `application:didFinishLaunchingWithOptions` method, replace any Bugsnag initialization code with the updated version:
+
+```diff
+- [BugsnagReactNative start];
++ [Bugsnag start];
+```
+
+Or if you need to customize your configuration:
+
+```diff
+- BugsnagConfiguration *config = [BugsnagConfiguration new];
+- config.releaseStage = @"beta";
+- [BugsnagReactNative startWithConfiguration:config];
++ BugsnagConfiguration *config = [BugsnagConfiguration loadConfig];
++ config.releaseStage = @"beta";
++ [Bugsnag startWithConfiguration:config];
+```
+
+See the new [Bugsnag Cocoa configuration guide](https://docs.bugsnag.com/platforms/ios/configuration-options/) for all available configuration options.
+
+If you have set your API key in `Info.plist`, the format has changed from a top level string to a dictionary.
+
+```
+- <key>BugsnagAPIKey</key>
+- <string>YOUR_API_KEY</string>
++ <key>bugsnag</key>
++ <dict>
++     <key>apiKey</key>
++     <string>YOUR_API_KEY</string>
++ </dict>
+```
+
+##### Android
+
+Replace the following import which will likely be in your `MainApplication.java` file:
+
+```diff
+- import com.bugsnag.android.BugsnagReactNative;
++ import com.bugsnag.android.Bugsnag;
+```
+
+Inside your application's `onCreate` method, replace any Bugsnag initialization code with the updated version:
+
+```diff
+- Configuration config = new Configuration();
+- config.setReleaseStage("beta");
+- BugsnagReactNative.startWithConfiguration(this /* app context */, config);
++ Configuration config = Configuration.load(this);
++ config.setReleaseStage("beta");
+Bugsnag.start(this /* app context */, config);
+```
+
+###### JS
+
+The JS interface has been updated to match our new notifier spec standard – that means it looks the same as our JS notifier, and uses the same names and concepts as our other platforms.
+
+When the JS layer loads, __the native layers and notifiers must already have been configured__. It you attempt to initialize the JS layer without having first configured the native layer, it __will not work__. This is new behaviour – previously we allowed configuration of the native layer to be done lazily and automatically, but this meant native errors on startup could be missed.
+
+Since the JS layer hooks in to an already running native layer, there are fewer configuration options available. Most configuration must be done in the native layer, with only a small subset of JS-specific options allowed in the JS layer.
```

Comment:
Move underneath proceeding snippet as it's referring to configuration, not starting?